### PR TITLE
Dev poisson ratio warning

### PR DESCRIPTION
**Pull request by armantekinalp**
- Soft changes are done for rod initialization. If user does not provides `shear modulus` then warning messages will be raised and `shear modulus` will be computed based on `Poisson Ratio`. 
- We changed the test scripts to test new warning messages.
- Old test scripts are updated such that warning messages won't be raised.
- Examples are updated such that warning messages won't be raised.

### CODE REVIEW

File: elastica/rod/factory_function.py [hunk 1]
```diff
@@ -216,7 +215,43 @@ def allocate(
...
+        else:
+            message = "Shear modulus or poisson ratio cannot be found in kwargs. Poisson ratio is taken as 0.5"
+            warnings.warn(message, category=UserWarning)
+            poisson_ratio = 0.5
+
+        shear_modulus = youngs_modulus / (poisson_ratio + 1.0)
+
+        message = (
+            "Shear modulus cannot be found in kwargs. \n"
+            "Poisson ratio "
```
Comment (nmnaughton) [inline elastica/rod/factory_function.py]: I would suggest the warning being:
 "The given Poisson ratio of "+ str(poisson_ratio) + " is used only to compute a shear modulus of " + str(shear_modulus) + ", " "using the equation: shear_modulus = youngs_modulus / (poisson_ratio + 1.0). Use of a Poisson ratio will be depreciated in a future release. It is encouraged that you discontinue using a Poisson ratio and instead directly provide the shear_modulus"

File: elastica/rod/factory_function.py [hunk 2]
```diff
@@ -216,7 +215,43 @@ def allocate(
         )
 
     # Shear/Stretch matrix
-    shear_modulus = youngs_modulus / (poisson_ratio + 1.0)
+    if kwargs.__contains__("shear_modulus"):
+        shear_modulus = kwargs.get("shear_modulus")
+        if kwargs.__contains__("poisson_ratio"):
+            poisson_ratio = kwargs.get("poisson_ratio")
+            message = (
+                " Poisson ratio ( "
```
Comment (nmnaughton) [inline elastica/rod/factory_function.py]: I would suggest:
"Both a Poisson ratio and a shear modulus are provided. The Poisson ratio is only used to compute a shear modulus so the provided Poisson ratio of ( "+ str(poisson_ratio)+ " ) is being ignored in favor of the provided shear modulus ( "+ str(shear_modulus)+ " ) \n"

**Comment by nmnaughton** [comment]
@armantekinalp  I made two suggestions to changing the warnings to hopefully make them a bit more clear. Let me know what you think.

**Comment by armantekinalp** [comment]
I made the changes and merging to the master.

**Pull request merged**